workspaces/analytics/src/eventbus.ts

```diff
@@ -14,6 +14,7 @@ export function newAnalyticsEventBus(
     emit: async (...events: TrackingEventBase<any>[]) => {
       const batchId = uuidv4();
       const context = await getContext(batchId);
+      // TODO - don't overwrite context
```

Comment:
as in Object.freeze?

Reply:
No, `events` contains a `context` key, but here we completely ignore it and overwrite it. We probably should merge these, but if we take a step back we should probably distinguish what we use context for (and what we use data for)

Data could be used as a "event specific" items, and context is something that should be sent with every event. Which would involve updating the typings (such that events, and UI events passed through event specific data, and context is applied always in the eventbus)

Reply:
Oh I see, yep that makes sense.